storage/storage_dest.go

```diff
@@ -272,7 +272,13 @@ func (s *storageImageDestination) putBlobToPendingFile(stream io.Reader, blobinf
 	if err != nil {
 		return private.UploadedBlob{}, fmt.Errorf("creating temporary file %q: %w", filename, err)
 	}
-	defer file.Close()
+	// since we are writing to this file, make sure we handle err on Close()
+	defer func() {
+		closeErr := file.Close()
+		if retErr == nil {
+			retErr = closeErr
+		}
+	}()
```

Comment:
This function should exit before updating `s.lockProtected` and the like.

Reply:
Done.